sql/aiven_extras.sql

```diff
@@ -410,4 +410,46 @@ CREATE OR REPLACE VIEW aiven_extras.pg_stat_replication AS
 
 PERFORM pg_catalog.set_config('search_path', old_path, true);
 END;
+
+
+DROP FUNCTION IF EXISTS aiven_extras.pg_create_publication;
+CREATE FUNCTION aiven_extras.pg_create_publication(
+    arg_publication_name TEXT,
+    arg_publish TEXT,
+    VARIADIC arg_tables TEXT[] DEFAULT ARRAY[]::TEXT[]
+)
+RETURNS VOID LANGUAGE plpgsql AS $$
+DECLARE
+  l_ident TEXT;
+  l_table_count INT;
+  l_tables_command TEXT;
+  l_parsed_ident TEXT[];
+  l_parsed_arg_tables TEXT[];
+BEGIN
+    l_table_count = array_length(arg_tables, 1);
+    IF l_table_count >= 1
+    THEN
+        l_parsed_arg_tables = ARRAY[]::TEXT[];
+        l_tables_command = 'CREATE PUBLICATION %I FOR TABLE ';
+        FOREACH l_ident IN ARRAY arg_tables LOOP
+            l_parsed_ident = parse_ident(l_ident);
+            ASSERT array_length(l_parsed_ident, 1) <= 2, 'Only simple table names or tables qualified with schema names allowed';
```

Comment:
This also works (didn't know about the `parse_ident` before). I would normally let Postgresql decide if it's valid or not with something like this:

```
SELECT
c.relname, n.nspname
FROM pg_class c
JOIN pg_namespace n ON n.OID=c.relnamespace
WHERE c.OID=l_ident::REGCLASS::INT AND relkind='r';
```

This makes sure that you have a table (not a partitioned table, or other stuff that's not supported according to docs):

>Only persistent base tables can be part of a publication. Temporary tables, unlogged tables, foreign tables, materialized views, regular views, and partitioned tables cannot be part of a publication.

Reply:
The point of `parse_ident` was actually to make this more usable from python. The schema.table string will end up double quoted here, and i'm using `parse_ident` to split the string into relevant parts then rejoin them where it makes sense. Any errors related to missing tables or mismatched table type are handled by the actual `CREATE PUBLICATION` statement, with the relevant error returned to the user